{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# IPython Jupyter Notebook for Analyzing 16S rRNA Sequence Data Using Mothur"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reference files for alignment can be found at https://www.mothur.org/wiki/Silva_reference_files \n",
    "### Taxonomy trainset files can be found at https://www.mothur.org/wiki/RDP_reference_files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Set location of (1) mothur application, (2) fastq files, (2) desired location for output, (4) number of processors of computer, (5) reference alignment, (6) fasta file for taxonomy trainset, (7) tax file for taxonomy trainset\n",
    "## Note: Windows users must include two backslashes in path names. This is not necessary for any other operating system."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "mothur_path = 'C:\\\\Users\\\\Haley\\\\Documents\\\\MiSeq\\\\MiSeq_SOP\\\\mothur\\\\mothur\\\\mothur.exe'\n",
    "inputDir = 'C:\\\\Users\\\\Haley\\\\Documents\\\\MiSeq\\\\MiSeq_SOP\\\\mothur\\\\mothur\\\\New_Data\\\\H1telson'\n",
    "outputDir = 'C:\\\\Users\\\\Haley\\\\Documents\\\\MiSeq\\\\MiSeq_SOP\\\\H1_Output'\n",
    "processors = 4\n",
    "reference = 'silva.bacteria.fasta'\n",
    "trainsetFasta = 'trainset9_032012.pds.fasta'\n",
    "trainsetTax = 'trainset9_032012.pds.tax'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Checking to make sure that the mothur application path and the input directory exist\n",
    "### If an error message appears, go back and check your path names to make sure they are correct"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "from colorama import Fore\n",
    "import os\n",
    "import sys\n",
    "\n",
    "if os.path.exists(mothur_path) == False:\n",
    "    print(Fore.RED + \"Error: Mothur path does not exist!\")\n",
    "\n",
    "if os.path.isdir(inputDir) == False:\n",
    "    print(Fore.RED + \"Error: Input directory does not exist!\")\n",
    "\n",
    "if sys.platform.startswith('win'):\n",
    "    slash = '\\\\'\n",
    "else:\n",
    "    slash = '/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Importing module necessary to run mothur in Jupyter Notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "from mothur_py import Mothur\n",
    "m = Mothur(mothur_path=mothur_path)\n",
    "m.verbosity=1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Set directories"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > set.dir(input=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson,output=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output)\n",
      "Mothur's directories:\n",
      "outputDir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\\n",
      "inputDir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\\n",
      "\n"
     ]
    }
   ],
   "source": [
    "m.set.dir(input=inputDir, output = outputDir)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Make file from fastq files containing name of sample, name of forward read and name of reverse read in separate columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > make.file(type=fastq)\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.files\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "m.make.file(type='fastq')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Make contigs from the two sets of reads for each sample and summarize them"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > make.contigs(file=stability.files,processors=4)\n",
      "Unable to open C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\stability.files. Trying input directory C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\stability.files.\n",
      "Unable to open C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\stability.files. Trying output directory C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.files.\n",
      "\n",
      "Using 4 processors.\n",
      "\n",
      ">>>>>\tProcessing file pair C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\H1ARIZONENSISSCORPTELSONE01_S30_L001_R1_001.FASTQ - C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\H1ARIZONENSISSCORPTELSONE01_S30_L001_R2_001.FASTQ (files 1 of 1)\t<<<<<\n",
      "Making contigs...\n",
      "1000\n",
      "1000\n",
      "1000\n",
      "1000\n",
      "2000\n",
      "2000\n",
      "2000\n",
      "2000\n",
      "3000\n",
      "3000\n",
      "3000\n",
      "3000\n",
      "4000\n",
      "4000\n",
      "4000\n",
      "4000\n",
      "5000\n",
      "5000\n",
      "5000\n",
      "5000\n",
      "6000\n",
      "6000\n",
      "6000\n",
      "6000\n",
      "7000\n",
      "7000\n",
      "7000\n",
      "7000\n",
      "8000\n",
      "8000\n",
      "8000\n",
      "8000\n",
      "9000\n",
      "9000\n",
      "9000\n",
      "9000\n",
      "10000\n",
      "10000\n",
      "10000\n",
      "10000\n",
      "11000\n",
      "11000\n",
      "11000\n",
      "11000\n",
      "12000\n",
      "12000\n",
      "12000\n",
      "12000\n",
      "13000\n",
      "13000\n",
      "13000\n",
      "13000\n",
      "13878\n",
      "13878\n",
      "13878\n",
      "13880\n",
      "Done.\n",
      "\n",
      "It took 41 secs to assemble 55514 reads.\n",
      "\n",
      "\n",
      "Group count: \n",
      "H1ARIZONENSISSCORPTELSONE01\t55514\n",
      "\n",
      "Total of all groups is 55514\n",
      "\n",
      "It took 41 secs to process 55514 sequences.\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.fasta\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.qual\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.scrap.contigs.fasta\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.scrap.contigs.qual\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.contigs.report\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.contigs.groups\n",
      "\n",
      "\n",
      "mothur > summary.seqs(fasta=current)\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.fasta as input file for the fasta parameter.\n",
      "\n",
      "Using 4 processors.\n",
      "\n",
      "\t\tStart\tEnd\tNBases\tAmbigs\tPolymer\tNumSeqs\n",
      "Minimum:\t1\t247\t247\t0\t3\t1\n",
      "2.5%-tile:\t1\t252\t252\t0\t4\t1388\n",
      "25%-tile:\t1\t253\t253\t0\t4\t13879\n",
      "Median: \t1\t253\t253\t0\t4\t27758\n",
      "75%-tile:\t1\t253\t253\t0\t5\t41636\n",
      "97.5%-tile:\t1\t281\t281\t6\t6\t54127\n",
      "Maximum:\t1\t501\t501\t57\t240\t55514\n",
      "Mean:\t1\t255\t255\t0\t4\n",
      "# of Seqs:\t55514\n",
      "\n",
      "It took 2 secs to summarize 55514 sequences.\n",
      "\n",
      "Output File Names:\n",
      " C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.summary\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "m.make.contigs(file='stability.files', processors = processors)\n",
    "m.summary.seqs(fasta='current')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## It's important to note the percent of sequences that are too long or that contain more than one ambiguous base call for the screening step that comes next."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Percent of all sequences with length > 275 bp: 4.16%\n",
      "\n",
      "Percent of all sequences with at least 1 ambiguous base call: 18.06%\n",
      "\n"
     ]
    }
   ],
   "source": [
    "with open(outputDir + slash + 'stability.trim.contigs.summary') as f :\n",
    "    lines = [l.strip() for l in f]\n",
    "lines = lines[1:]\n",
    "\n",
    "lengths = [s.split('\\t')[3] for s in lines]\n",
    "lengths = [int(i) for i in lengths]\n",
    "long = 0\n",
    "for i in lengths:\n",
    "    if i >= 275:\n",
    "        long += 1\n",
    "print(\"\\nPercent of all sequences with length > 275 bp: \" + str(round((long/len(lengths)*100),2)) + '%')\n",
    "\n",
    "ambigs = [s.split('\\t')[4] for s in lines]\n",
    "ambigs = [int(i) for i in ambigs]\n",
    "num = 0\n",
    "for i in ambigs:\n",
    "    if i > 0:\n",
    "        num += 1\n",
    "print(\"\\nPercent of all sequences with at least 1 ambiguous base call: \" + str(round((num/len(ambigs)*100),2)) + '%\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Set Parameters For Screening Step\n",
    "### If the percent of sequences with more than one ambigous base calls is too high, then set maxambig to an appropriate number other than 0.\n",
    "    \n",
    "### If the percent of sequences longer than 275 is too high, then reset maxlength to a more appropriate number."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "maxambig=0\n",
    "maxlength=275"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Screening out undesired sequences based on parameters set in previous step."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > screen.seqs(fasta=current,group=current,maxambig=0,maxlength=275)\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.fasta as input file for the fasta parameter.\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.contigs.groups as input file for the group parameter.\n",
      "\n",
      "Using 4 processors.\n",
      "1000\n",
      "1000\n",
      "1000\n",
      "1000\n",
      "2000\n",
      "2000\n",
      "2000\n",
      "2000\n",
      "3000\n",
      "3000\n",
      "3000\n",
      "3000\n",
      "4000\n",
      "4000\n",
      "4000\n",
      "4000\n",
      "5000\n",
      "5000\n",
      "5000\n",
      "5000\n",
      "6000\n",
      "6000\n",
      "6000\n",
      "6000\n",
      "7000\n",
      "7000\n",
      "7000\n",
      "7000\n",
      "8000\n",
      "8000\n",
      "8000\n",
      "8000\n",
      "9000\n",
      "9000\n",
      "9000\n",
      "9000\n",
      "10000\n",
      "10000\n",
      "10000\n",
      "10000\n",
      "11000\n",
      "11000\n",
      "11000\n",
      "11000\n",
      "12000\n",
      "12000\n",
      "12000\n",
      "12000\n",
      "13000\n",
      "13000\n",
      "13000\n",
      "13000\n",
      "13878\n",
      "13878\n",
      "13878\n",
      "13880\n",
      "\n",
      "It took 1 secs to screen 55514 sequences, removed 11847.\n",
      "\n",
      "/******************************************/\n",
      "Running command: remove.seqs(accnos=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.bad.accnos, group=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.contigs.groups)\n",
      "Removed 11847 sequences from your group file.\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.contigs.pick.groups\n",
      "\n",
      "/******************************************/\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.fasta\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.bad.accnos\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.contigs.good.groups\n",
      "\n",
      "\n",
      "It took 2 secs to screen 55514 sequences.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "m.screen.seqs(fasta='current',group = 'current', maxambig=maxambig, maxlength=maxlength)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Merge Identical Sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > unique.seqs(fasta=current)\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.fasta as input file for the fasta parameter.\n",
      "1000\t582\n",
      "2000\t1034\n",
      "3000\t1443\n",
      "4000\t1844\n",
      "5000\t2186\n",
      "6000\t2570\n",
      "7000\t2923\n",
      "8000\t3260\n",
      "9000\t3584\n",
      "10000\t3926\n",
      "11000\t4227\n",
      "12000\t4528\n",
      "13000\t4806\n",
      "14000\t5076\n",
      "15000\t5366\n",
      "16000\t5642\n",
      "17000\t5915\n",
      "18000\t6175\n",
      "19000\t6467\n",
      "20000\t6739\n",
      "21000\t7012\n",
      "22000\t7272\n",
      "23000\t7528\n",
      "24000\t7765\n",
      "25000\t8014\n",
      "26000\t8307\n",
      "27000\t8550\n",
      "28000\t8803\n",
      "29000\t9045\n",
      "30000\t9270\n",
      "31000\t9505\n",
      "32000\t9757\n",
      "33000\t9998\n",
      "34000\t10250\n",
      "35000\t10487\n",
      "36000\t10707\n",
      "37000\t10934\n",
      "38000\t11180\n",
      "39000\t11414\n",
      "40000\t11633\n",
      "41000\t11876\n",
      "42000\t12105\n",
      "43000\t12332\n",
      "43667\t12486\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.names\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.fasta\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "m.unique.seqs(fasta='current')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Generate count file with names of representative sequences and number of sequences in that group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > count.seqs(name=current,group=current)\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.contigs.good.groups as input file for the group parameter.\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.names as input file for the name parameter.\n",
      "\n",
      "It took 1 secs to create a table for 43667 sequences.\n",
      "\n",
      "Total number of sequences: 43667\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.count_table\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "m.count.seqs(name='current', group='current')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Number of unique sequences out of total sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "12486 out of 43667 sequences are unique.\n"
     ]
    }
   ],
   "source": [
    "with open(outputDir + slash + 'stability.trim.contigs.good.count_table') as f:\n",
    "    lines = [l.strip() for l in f]\n",
    "lines = lines[1:]\n",
    "\n",
    "seqs = [s.split('\\t')[1] for s in lines]\n",
    "seqs = [int(i) for i in seqs]\n",
    "\n",
    "uniques = len(lines)\n",
    "total = sum(seqs)\n",
    "\n",
    "print('\\n' + str(uniques) + \" out of \" + str(total) + \" sequences are unique.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Align Sequences to Reference Alignment. Start and end correspond to the V4 region of 16S rRNA gene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > pcr.seqs(fasta=silva.bacteria.fasta,start=11894,end=25319,keepdots=F)\n",
      "Unable to open C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\silva.bacteria.fasta. Trying input directory C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\silva.bacteria.fasta.\n",
      "Unable to open C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\silva.bacteria.fasta. Trying output directory C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\silva.bacteria.fasta.\n",
      "Unable to open C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\silva.bacteria.fasta. Trying default C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\silva.bacteria.fasta.\n",
      "\n",
      "Using 4 processors.\n",
      "1000\n",
      "1000\n",
      "1000\n",
      "1000\n",
      "2000\n",
      "2000\n",
      "2000\n",
      "2000\n",
      "3000\n",
      "3000\n",
      "3000\n",
      "3000\n",
      "3739\n",
      "3739\n",
      "3739\n",
      "3739\n",
      "[NOTE]: no sequences were bad, removing C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\silva.bacteria.bad.accnos\n",
      "\n",
      "It took 34 secs to screen 14956 sequences.\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\silva.bacteria.pcr.fasta\n",
      "\n",
      "\n",
      "\n",
      "mothur > align.seqs(fasta=stability.trim.contigs.good.unique.fasta,reference=silva.bacteria.pcr.fasta)\n",
      "Unable to open C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\silva.bacteria.pcr.fasta. Trying input directory C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\silva.bacteria.pcr.fasta.\n",
      "Unable to open C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\silva.bacteria.pcr.fasta. Trying output directory C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\silva.bacteria.pcr.fasta.\n",
      "Unable to open stability.trim.contigs.good.unique.fasta. Trying input directory C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\stability.trim.contigs.good.unique.fasta.\n",
      "Unable to open C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\stability.trim.contigs.good.unique.fasta. Trying output directory C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.fasta.\n",
      "\n",
      "Using 4 processors.\n",
      "\n",
      "Reading in the C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\silva.bacteria.pcr.fasta template sequences...\tDONE.\n",
      "It took 10 to read  14956 sequences.\n",
      "Aligning sequences from C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.fasta ...\n",
      "1000\n",
      "1000\n",
      "1000\n",
      "1000\n",
      "2000\n",
      "2000\n",
      "2000\n",
      "2000\n",
      "3000\n",
      "3000\n",
      "3000\n",
      "3000\n",
      "3121\n",
      "3123\n",
      "3121\n",
      "3121\n",
      "It took 20 secs to align 12486 sequences.\n",
      "\n",
      "[WARNING]: 46 of your sequences generated alignments that eliminated too many bases, a list is provided in C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.flip.accnos.\n",
      "[NOTE]: 10 of your sequences were reversed to produce a better alignment.\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.align\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.align.report\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.flip.accnos\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "m.pcr.seqs(fasta=reference, start=11894, end=25319, keepdots='F')\n",
    "m.align.seqs(fasta='stability.trim.contigs.good.unique.fasta', reference= 'silva.bacteria.pcr.fasta')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Re-screen to Remove Sequences With Indels and Homopolymers Longer than 8 Bases\n",
    "#### V4 region of 16S does not include homopolymers longer than 8 bases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > summary.seqs(fasta=current,count=current)\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.count_table as input file for the count parameter.\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.align as input file for the fasta parameter.\n",
      "\n",
      "Using 4 processors.\n",
      "\n",
      "\t\tStart\tEnd\tNBases\tAmbigs\tPolymer\tNumSeqs\n",
      "Minimum:\t1\t1964\t1\t0\t1\t1\n",
      "2.5%-tile:\t1968\t11550\t252\t0\t4\t1092\n",
      "25%-tile:\t1968\t11550\t253\t0\t4\t10917\n",
      "Median: \t1968\t11550\t253\t0\t4\t21834\n",
      "75%-tile:\t1968\t11550\t253\t0\t5\t32751\n",
      "97.5%-tile:\t1968\t11550\t254\t0\t6\t42576\n",
      "Maximum:\t13425\t13425\t273\t0\t8\t43667\n",
      "Mean:\t1977\t11552\t252\t0\t4\n",
      "# of unique seqs:\t12486\n",
      "total # of seqs:\t43667\n",
      "\n",
      "It took 8 secs to summarize 43667 sequences.\n",
      "\n",
      "Output File Names:\n",
      " C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.summary\n",
      "\n",
      "\n",
      "mothur > screen.seqs(fasta=current,count=current,summary=current,start=1968,end=11550,maxhomop=8)\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.count_table as input file for the count parameter.\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.align as input file for the fasta parameter.\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.summary as input file for the summary parameter.\n",
      "\n",
      "Using 4 processors.\n",
      "1000\n",
      "1000\n",
      "1000\n",
      "1000\n",
      "2000\n",
      "2000\n",
      "2000\n",
      "2000\n",
      "3000\n",
      "3000\n",
      "3000\n",
      "3000\n",
      "3123\n",
      "3121\n",
      "3121\n",
      "3121\n",
      "\n",
      "It took 4 secs to screen 12486 sequences, removed 127.\n",
      "\n",
      "/******************************************/\n",
      "Running command: remove.seqs(accnos=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.bad.accnos, count=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.count_table)\n",
      "Removed 167 sequences from your count file.\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.pick.count_table\n",
      "\n",
      "/******************************************/\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.summary\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.align\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.bad.accnos\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.good.count_table\n",
      "\n",
      "\n",
      "It took 8 secs to screen 12486 sequences.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "m.summary.seqs(fasta='current', count = 'current')\n",
    "m.screen.seqs(fasta='current', count='current', summary='current', start=1968, end=11550, maxhomop=8)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Filter sequences to remove columns that only contain gaps and extra gaps at ends"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > filter.seqs(fasta=current,vertical=T,trump=.)\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.align as input file for the fasta parameter.\n",
      "\n",
      "Using 4 processors.\n",
      "Creating Filter... \n",
      "1000\n",
      "1000\n",
      "1000\n",
      "1000\n",
      "2000\n",
      "2000\n",
      "2000\n",
      "2000\n",
      "3000\n",
      "3000\n",
      "3000\n",
      "3000\n",
      "3089\n",
      "3092\n",
      "3089\n",
      "3089\n",
      "It took 5 secs to create filter for 12359 sequences.\n",
      "\n",
      "\n",
      "Running Filter... \n",
      "1000\n",
      "1000\n",
      "1000\n",
      "1000\n",
      "2000\n",
      "2000\n",
      "2000\n",
      "2000\n",
      "3000\n",
      "3000\n",
      "3000\n",
      "3000\n",
      "3089\n",
      "3089\n",
      "3092\n",
      "3089\n",
      "It took 4 secs to filter 12359 sequences.\n",
      "\n",
      "\n",
      "\n",
      "Length of filtered alignment: 415\n",
      "Number of columns removed: 13010\n",
      "Length of the original alignment: 13425\n",
      "Number of sequences used to construct filter: 12359\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.filter\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.fasta\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "m.filter.seqs(fasta='current', vertical='T', trump='.')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Merge any new duplicate sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > unique.seqs(fasta=current,count=current)\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.good.count_table as input file for the count parameter.\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.fasta as input file for the fasta parameter.\n",
      "1000\t999\n",
      "2000\t1999\n",
      "3000\t2998\n",
      "4000\t3998\n",
      "5000\t4997\n",
      "6000\t5997\n",
      "7000\t6997\n",
      "8000\t7996\n",
      "9000\t8996\n",
      "10000\t9995\n",
      "11000\t10995\n",
      "12000\t11992\n",
      "12359\t12351\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.count_table\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.fasta\n",
      "\n",
      "\n",
      "mothur > summary.seqs(fasta=current,count=current)\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.count_table as input file for the count parameter.\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.fasta as input file for the fasta parameter.\n",
      "\n",
      "Using 4 processors.\n",
      "\n",
      "\t\tStart\tEnd\tNBases\tAmbigs\tPolymer\tNumSeqs\n",
      "Minimum:\t1\t415\t247\t0\t3\t1\n",
      "2.5%-tile:\t1\t415\t252\t0\t4\t1088\n",
      "25%-tile:\t1\t415\t253\t0\t4\t10876\n",
      "Median: \t1\t415\t253\t0\t4\t21751\n",
      "75%-tile:\t1\t415\t253\t0\t5\t32626\n",
      "97.5%-tile:\t1\t415\t254\t0\t6\t42413\n",
      "Maximum:\t1\t415\t273\t0\t8\t43500\n",
      "Mean:\t1\t415\t252\t0\t4\n",
      "# of unique seqs:\t12351\n",
      "total # of seqs:\t43500\n",
      "\n",
      "It took 0 secs to summarize 43500 sequences.\n",
      "\n",
      "Output File Names:\n",
      " C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.summary\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "m.unique.seqs(fasta='current', count='current')\n",
    "m.summary.seqs(fasta='current', count='current')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Pre-cluster (merge) sequences within 2 bases of each other"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > pre.cluster(fasta=current,count=current,diffs=2)\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.count_table as input file for the count parameter.\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.fasta as input file for the fasta parameter.\n",
      "\n",
      "Using 4 processors.\n",
      "Reducing processors to 1.\n",
      "\n",
      "Processing group H1ARIZONENSISSCORPTELSONE01:\n",
      "H1ARIZONENSISSCORPTELSONE01\t0\t11695\t656\n",
      "H1ARIZONENSISSCORPTELSONE01\t100\t6147\t6204\n",
      "H1ARIZONENSISSCORPTELSONE01\t200\t5156\t7195\n",
      "H1ARIZONENSISSCORPTELSONE01\t300\t4445\t7906\n",
      "H1ARIZONENSISSCORPTELSONE01\t400\t4039\t8312\n",
      "H1ARIZONENSISSCORPTELSONE01\t500\t3721\t8630\n",
      "H1ARIZONENSISSCORPTELSONE01\t600\t3482\t8869\n",
      "H1ARIZONENSISSCORPTELSONE01\t700\t3328\t9023\n",
      "H1ARIZONENSISSCORPTELSONE01\t800\t3180\t9171\n",
      "H1ARIZONENSISSCORPTELSONE01\t900\t3014\t9337\n",
      "H1ARIZONENSISSCORPTELSONE01\t1000\t2905\t9446\n",
      "H1ARIZONENSISSCORPTELSONE01\t1100\t2796\t9555\n",
      "H1ARIZONENSISSCORPTELSONE01\t1200\t2723\t9628\n",
      "H1ARIZONENSISSCORPTELSONE01\t1300\t2664\t9687\n",
      "H1ARIZONENSISSCORPTELSONE01\t1400\t2630\t9721\n",
      "H1ARIZONENSISSCORPTELSONE01\t1500\t2568\t9783\n",
      "H1ARIZONENSISSCORPTELSONE01\t1600\t2544\t9807\n",
      "H1ARIZONENSISSCORPTELSONE01\t1700\t2511\t9840\n",
      "H1ARIZONENSISSCORPTELSONE01\t1800\t2485\t9866\n",
      "H1ARIZONENSISSCORPTELSONE01\t1900\t2474\t9877\n",
      "H1ARIZONENSISSCORPTELSONE01\t2000\t2454\t9897\n",
      "H1ARIZONENSISSCORPTELSONE01\t2100\t2452\t9899\n",
      "H1ARIZONENSISSCORPTELSONE01\t2200\t2418\t9933\n",
      "H1ARIZONENSISSCORPTELSONE01\t2300\t2345\t10006\n",
      "H1ARIZONENSISSCORPTELSONE01\t2400\t2333\t10018\n",
      "H1ARIZONENSISSCORPTELSONE01\t2500\t2323\t10028\n",
      "H1ARIZONENSISSCORPTELSONE01\t2600\t2312\t10039\n",
      "H1ARIZONENSISSCORPTELSONE01\t2700\t2282\t10069\n",
      "H1ARIZONENSISSCORPTELSONE01\t2800\t2272\t10079\n",
      "H1ARIZONENSISSCORPTELSONE01\t2900\t2254\t10097\n",
      "H1ARIZONENSISSCORPTELSONE01\t3000\t2249\t10102\n",
      "H1ARIZONENSISSCORPTELSONE01\t3100\t2243\t10108\n",
      "H1ARIZONENSISSCORPTELSONE01\t3200\t2230\t10121\n",
      "H1ARIZONENSISSCORPTELSONE01\t3300\t2218\t10133\n",
      "H1ARIZONENSISSCORPTELSONE01\t3400\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t3500\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t3600\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t3700\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t3800\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t3900\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t4000\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t4100\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t4200\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t4300\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t4400\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t4500\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t4600\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t4700\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t4800\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t4900\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t5000\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t5100\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t5200\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t5300\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t5400\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t5500\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t5600\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t5700\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t5800\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t5900\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t6000\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t6100\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t6200\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t6300\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t6400\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t6500\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t6600\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t6700\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t6800\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t6900\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t7000\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t7100\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t7200\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t7300\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t7400\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t7500\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t7600\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t7700\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t7800\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t7900\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t8000\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t8100\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t8200\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t8300\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t8400\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t8500\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t8600\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t8700\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t8800\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t8900\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t9000\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t9100\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t9200\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t9300\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t9400\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t9500\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t9600\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t9700\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t9800\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t9900\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t10000\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t10100\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t10200\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t10300\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t10400\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t10500\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t10600\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t10700\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t10800\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t10900\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t11000\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t11100\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t11200\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t11300\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t11400\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t11500\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t11600\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t11700\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t11800\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t11900\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t12000\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t12100\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t12200\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t12300\t2186\t10165\n",
      "H1ARIZONENSISSCORPTELSONE01\t12351\t2186\t10165\n",
      "Total number of sequences before pre.cluster was 12351.\n",
      "pre.cluster removed 10165 sequences.\n",
      "\n",
      "It took 3 secs to cluster 12351 sequences.\n",
      "It took 5 secs to run pre.cluster.\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.fasta\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.count_table\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.H1ARIZONENSISSCORPTELSONE01.map\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "m.pre.cluster(fasta='current', count = 'current', diffs = 2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Number of unique sequences out of total sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "2186 out of 43500 sequences are unique.\n"
     ]
    }
   ],
   "source": [
    "with open(outputDir + slash + 'stability.trim.contigs.good.unique.good.filter.unique.precluster.count_table') as f:\n",
    "    lines = [l.strip() for l in f]\n",
    "lines = lines[1:]\n",
    "\n",
    "seqs = [s.split('\\t')[1] for s in lines]\n",
    "seqs = [int(i) for i in seqs]\n",
    "\n",
    "uniques = len(lines)\n",
    "total = sum(seqs)\n",
    "\n",
    "print('\\n' + str(uniques) + \" out of \" + str(total) + \" sequences are unique.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Search for and remove chimeric sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > chimera.uchime(fasta=current,count=current,dereplicate=t)\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.count_table as input file for the count parameter.\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.fasta as input file for the fasta parameter.\n",
      "\n",
      "Using 4 processors.\n",
      "\n",
      "uchime by Robert C. Edgar\n",
      "http://drive5.com/uchime\n",
      "This code is donated to the public domain.\n",
      "\n",
      "Checking sequences from C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.fasta ...\n",
      "uchime v4.2.40\n",
      "by Robert C. Edgar\n",
      "http://drive5.com/uchime\n",
      "This code is donated to the public domain.\n",
      "\n",
      "00:00    0.1% Reading C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.temp00:00    0.1% Reading C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.temp00:00  100.0% Reading C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.temp\n",
      "00:00 2186 sequences                                                                                                                          \n",
      "00:00    0.0% 0/0 chimeras found (0.0%)00:01    7.6% 0/166 chimeras found (0.0%)00:02   13.1% 0/285 chimeras found (0.0%)00:03   17.4% 0/380 chimeras found (0.0%)00:04   20.8% 0/454 chimeras found (0.0%)00:05   24.5% 2/534 chimeras found (0.4%)00:06   28.3% 2/617 chimeras found (0.3%)00:07   31.1% 3/678 chimeras found (0.4%)00:08   33.9% 4/741 chimeras found (0.5%)00:09   36.9% 5/805 chimeras found (0.6%)00:10   39.2% 6/855 chimeras found (0.7%)00:11   41.3% 6/901 chimeras found (0.7%)00:12   43.6% 7/953 chimeras found (0.7%)00:13   45.8% 8/1001 chimeras found (0.8%)00:14   47.4% 8/1036 chimeras found (0.8%)00:15   50.0% 10/1091 chimeras found (0.9%)00:16   52.4% 11/1144 chimeras found (1.0%)00:17   54.5% 13/1191 chimeras found (1.1%)00:18   55.4% 15/1211 chimeras found (1.2%)00:19   57.2% 15/1249 chimeras found (1.2%)00:20   59.0% 16/1289 chimeras found (1.2%)00:21   60.4% 17/1319 chimeras found (1.3%)00:22   61.2% 18/1336 chimeras found (1.3%)00:23   62.1% 19/1356 chimeras found (1.4%)00:24   63.7% 20/1391 chimeras found (1.4%)00:25   65.1% 21/1422 chimeras found (1.5%)00:26   66.9% 22/1461 chimeras found (1.5%)00:27   68.5% 25/1497 chimeras found (1.7%)00:28   70.5% 26/1540 chimeras found (1.7%)00:29   73.0% 26/1595 chimeras found (1.6%)00:30   74.2% 27/1622 chimeras found (1.7%)00:31   74.9% 28/1636 chimeras found (1.7%)00:32   75.5% 29/1650 chimeras found (1.8%)00:33   76.4% 31/1669 chimeras found (1.9%)00:34   76.9% 32/1681 chimeras found (1.9%)00:35   77.7% 33/1697 chimeras found (1.9%)00:36   79.0% 34/1725 chimeras found (2.0%)00:37   80.0% 34/1747 chimeras found (1.9%)00:38   81.1% 34/1771 chimeras found (1.9%)00:39   82.5% 34/1803 chimeras found (1.9%)00:40   83.8% 36/1830 chimeras found (2.0%)00:41   84.8% 36/1852 chimeras found (1.9%)00:42   85.9% 36/1877 chimeras found (1.9%)00:43   87.6% 37/1913 chimeras found (1.9%)00:44   89.4% 42/1954 chimeras found (2.1%)00:45   90.9% 42/1985 chimeras found (2.1%)00:46   91.9% 43/2007 chimeras found (2.1%)00:47   93.5% 43/2042 chimeras found (2.1%)00:48   94.5% 43/2065 chimeras found (2.1%)00:49   95.5% 44/2087 chimeras found (2.1%)00:50   97.1% 44/2121 chimeras found (2.1%)00:51   98.8% 45/2158 chimeras found (2.1%)00:52  100.0% 45/2184 chimeras found (2.1%)00:52  100.0% 45/2185 chimeras found (2.1%)\n",
      "Reducing processors to 1.\n",
      "\n",
      "It took 52 secs to check 2186 sequences from group H1ARIZONENSISSCORPTELSONE01.\n",
      "It took 53 secs to check 2186 sequences.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.uchime.pick.count_table\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.uchime.chimeras\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.uchime.accnos\n",
      "\n",
      "\n",
      "mothur > remove.seqs(fasta=current,accnos=current)\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.uchime.accnos as input file for the accnos parameter.\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.fasta as input file for the fasta parameter.\n",
      "[WARNING]: This command can take a namefile and you did not provide one. The current namefile is C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.names which seems to match C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.fasta.\n",
      "Removed 45 sequences from your fasta file.\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.fasta\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "m.chimera.uchime(fasta = 'current', count = 'current', dereplicate = 't')\n",
    "m.remove.seqs(fasta='current', accnos='current')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Classify sequences based on taxonomy to remove undesirables \n",
    "#### Sequences that do not belong to the domains Bacteria or Archaea are removed\n",
    "#### Select which taxon to remove"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "taxon='Chloroplast-Mitochondria-unknown-Eukaryota'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > classify.seqs(fasta=current,count=current,reference=trainset9_032012.pds.fasta,taxonomy=trainset9_032012.pds.tax,cutoff=80)\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.uchime.pick.count_table as input file for the count parameter.\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.fasta as input file for the fasta parameter.\n",
      "\n",
      "Using 4 processors.\n",
      "Unable to open C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\trainset9_032012.pds.fasta. Trying input directory C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\trainset9_032012.pds.fasta.\n",
      "Unable to open C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\trainset9_032012.pds.fasta. Trying output directory C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\trainset9_032012.pds.fasta.\n",
      "Unable to open C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\trainset9_032012.pds.fasta. Trying default C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\trainset9_032012.pds.fasta.\n",
      "Unable to open C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\trainset9_032012.pds.tax. Trying input directory C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\trainset9_032012.pds.tax.\n",
      "Unable to open C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\New_Data\\H1telson\\trainset9_032012.pds.tax. Trying output directory C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\trainset9_032012.pds.tax.\n",
      "Unable to open C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\trainset9_032012.pds.tax. Trying default C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\mothur\\mothur\\trainset9_032012.pds.tax.\n",
      "Reading template taxonomy...     DONE.\n",
      "Reading template probabilities...     DONE.\n",
      "It took 33 seconds get probabilities. \n",
      "Classifying sequences from C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.fasta ...\n",
      "[WARNING]: M01676_88_000000000-BW6W9_1_1105_23838_17889 could not be classified. You can use the remove.lineage command with taxon=unknown; to remove such sequences.\n",
      "100\n",
      "100\n",
      "100\n",
      "100\n",
      "200\n",
      "200\n",
      "200\n",
      "200\n",
      "[WARNING]: M01676_88_000000000-BW6W9_1_2109_28437_12138 could not be classified. You can use the remove.lineage command with taxon=unknown; to remove such sequences.\n",
      "300\n",
      "300\n",
      "300\n",
      "300\n",
      "400\n",
      "[WARNING]: M01676_88_000000000-BW6W9_1_2108_15942_18187 could not be classified. You can use the remove.lineage command with taxon=unknown; to remove such sequences.\n",
      "400\n",
      "400\n",
      "[WARNING]: M01676_88_000000000-BW6W9_1_1101_21994_10916 could not be classified. You can use the remove.lineage command with taxon=unknown; to remove such sequences.\n",
      "400\n",
      "[WARNING]: M01676_88_000000000-BW6W9_1_1104_8327_7488 could not be classified. You can use the remove.lineage command with taxon=unknown; to remove such sequences.\n",
      "500\n",
      "500\n",
      "500\n",
      "500\n",
      "535\n",
      "535\n",
      "535\n",
      "536\n",
      "It took 17 secs to classify 2141 sequences.\n",
      "\n",
      "\n",
      "It took 17 secs to classify 2141 sequences.\n",
      "\n",
      "\n",
      "It took 0 secs to create the summary file for 2141 sequences.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.taxonomy\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.tax.summary\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "m.classify.seqs(fasta='current', count='current', reference= trainsetFasta, taxonomy= trainsetTax, cutoff=80)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > remove.lineage(fasta=current,count=current,taxonomy=current,taxon=Chloroplast-Mitochondria-unknown-Eukaryota)\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.uchime.pick.count_table as input file for the count parameter.\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.fasta as input file for the fasta parameter.\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.taxonomy as input file for the taxonomy parameter.\n",
      "\n",
      "[NOTE]: The count file should contain only unique names, so mothur assumes your fasta, list and taxonomy files also contain only uniques.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.taxonomy\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.uchime.pick.pick.count_table\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "m.remove.lineage(fasta='current', count='current', taxonomy='current', taxon=taxon)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Cluster sequences and generate distance matrix and OTUs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > cluster.split(fasta=current,count=current,taxonomy=current,splitmethod=classify,taxlevel=4,cutoff=0.03)\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.uchime.pick.pick.count_table as input file for the count parameter.\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta as input file for the fasta parameter.\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.taxonomy as input file for the taxonomy parameter.\n",
      "\n",
      "Using 4 processors.\n",
      "Splitting the file...\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.0.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "200\t0\t216\n",
      "100\t0\t332\n",
      "\n",
      "It took 0 secs to find distances for 209 sequences. 1492 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.0.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.1.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 1 secs to find distances for 83 sequences. 1538 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.1.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.2.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "400\t0\t28\n",
      "0\t0\t0\n",
      "100\t0\t112\n",
      "700\t0\t220\n",
      "200\t0\t406\n",
      "600\t0\t588\n",
      "500\t0\t875\n",
      "300\t0\t759\n",
      "\n",
      "It took 0 secs to find distances for 794 sequences. 6603 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.2.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.3.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "200\t0\t21\n",
      "100\t0\t17\n",
      "\n",
      "It took 0 secs to find distances for 269 sequences. 211 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.3.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.4.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 54 sequences. 54 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.4.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.5.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 14 sequences. 8 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.5.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.6.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 17 sequences. 4 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.6.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.7.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 9 sequences. 2 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.7.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.8.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 31 sequences. 124 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.8.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.9.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 9 sequences. 9 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.9.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.10.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 12 sequences. 13 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.10.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.11.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 22 sequences. 17 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.11.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.12.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 50 sequences. 70 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.12.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.13.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "100\t0\t11\n",
      "\n",
      "It took 0 secs to find distances for 109 sequences. 199 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.13.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.14.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 26 sequences. 5 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.14.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.15.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 58 sequences. 264 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.15.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.16.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 2 sequences. 1 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.16.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.17.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 30 sequences. 53 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.17.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.18.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 1 secs to find distances for 15 sequences. 14 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.18.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.19.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 28 sequences. 17 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.19.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.20.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 27 sequences. 15 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.20.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.21.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 10 sequences. 16 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.21.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.22.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 2 sequences. 0 distances below cutoff 0.03.\n",
      "\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.22.dist is blank. This can result if there are no distances below your cutoff.\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.22.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.23.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 14 sequences. 5 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.23.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.24.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 16 sequences. 45 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.24.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.25.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 2 sequences. 1 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.25.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.26.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 23 sequences. 153 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.26.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.27.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 12 sequences. 1 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.27.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.28.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 26 sequences. 10 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.28.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.29.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 4 sequences. 1 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.29.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.30.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 3 sequences. 0 distances below cutoff 0.03.\n",
      "\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.30.dist is blank. This can result if there are no distances below your cutoff.\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.30.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.31.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 2 sequences. 0 distances below cutoff 0.03.\n",
      "\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.31.dist is blank. This can result if there are no distances below your cutoff.\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.31.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.32.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 7 sequences. 1 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.32.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.33.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 8 sequences. 0 distances below cutoff 0.03.\n",
      "\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.33.dist is blank. This can result if there are no distances below your cutoff.\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.33.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.34.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 13 sequences. 36 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.34.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.35.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 8 sequences. 3 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.35.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.36.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 4 sequences. 2 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.36.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.37.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 8 sequences. 5 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.37.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.38.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 3 sequences. 1 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.38.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.39.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 16 sequences. 44 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.39.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.40.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 8 sequences. 4 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.40.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.41.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 4 sequences. 3 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.41.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.42.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 5 sequences. 4 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.42.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.43.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 3 sequences. 1 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.43.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.44.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 9 sequences. 7 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.44.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.45.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 2 sequences. 0 distances below cutoff 0.03.\n",
      "\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.45.dist is blank. This can result if there are no distances below your cutoff.\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.45.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.46.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 4 sequences. 2 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.46.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.47.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 3 sequences. 2 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.47.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.48.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 2 sequences. 1 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.48.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.49.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 2 sequences. 1 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.49.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.50.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 4 sequences. 0 distances below cutoff 0.03.\n",
      "\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.50.dist is blank. This can result if there are no distances below your cutoff.\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.50.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.51.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 2 sequences. 1 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.51.dist\n",
      "\n",
      "/******************************************/\n",
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.52.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 4 sequences. 1 distances below cutoff 0.03.\n",
      "\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.52.dist\n",
      "\n",
      "/******************************************/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: dist.seqs(fasta=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.53.temp, processors=4, cutoff=0.03, outputdir=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\)\n",
      "\n",
      "Using 4 processors.\n",
      "/******************************************/\n",
      "\n",
      "Sequence\tTime\tNum_Dists_Below_Cutoff\n",
      "0\t0\t0\n",
      "\n",
      "It took 0 secs to find distances for 2 sequences. 0 distances below cutoff 0.03.\n",
      "\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.53.dist is blank. This can result if there are no distances below your cutoff.\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.53.dist\n",
      "\n",
      "It took 40 seconds to split the distance file.\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.1.dist\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.13.dist\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.4.dist\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.34.dist\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.20.dist\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.9.dist\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.23.dist\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.40.dist\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.36.dist\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.29.dist\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.32.dist\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.51.dist\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.49.dist\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.51.dist\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.48.dist\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.15.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "0\t702\t0\t1\t0\t1\t0\t0.998578\t1\t0.998578\t0\t702\t0\t1\t0\t0\t\n",
      "\n",
      "0\t1\t0\t0.998578\t1\t0.998578\t0\t0\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.32.dist\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.16.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "0\t702\t0\t1\t0\t1\t0\t0.998578\t1\t0.998578\t0\t0\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.43.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "0\t902\t0\t1\t0\t1\t0\t0.998893\t1\t0.998893\t0\t0\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.29.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "207\t4104\t3\t57\t0.784091\t0.99927\t0.985714\t0.986301\t0.985714\t0.986273\t0.872605\t0.873418\t\n",
      "\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "0\t702\t0\t1\t0\t1\t0\t0.998578\t1\t0.998578\t0\t0\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.38.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "0\t740\t0\t1\t0\t1\t0\t\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.8.dist\n",
      "0\t779\t0\t1\t0\t0.99865\t1\t1\t0\t0.99865\t0.998718\t1\t0\t0.998718\t0\t\n",
      "\n",
      "0\t0\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.36.dist\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.25.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "2\t778\t0\t0\t1\t1\t1\t1\t1\t1\t1\t1\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.40.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "4\t942\t0\t0\t1\t1\t1\t1\t1\t1\t1\t1\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.23.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "120\t2086\t1\t4\t0.967742\t0.999521\t0.991736\t0.998086\t0.991736\t0.997739\t0.978477\t0.979592\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.24.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "0\t702\t0\t1\t0\t1\t0\t0.998578\t1\t0.998578\t0\t0\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.46.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "5\t1220\t0\t0\t1\t1\t1\t1\t1\t1\t1\t1\t\n",
      "\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "0\t740\t0\t1\t0\t1\t0\t0.99865\t1\t0.99865\t0\t0\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.9.dist\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.7.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "45\t1278\t3\t0\t1\t0.997658\t0.9375\t1\t0.9375\t0.997738\t0.967111\t0.967742\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.11.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "1\t778\t0\t1\t0.5\t1\t1\t0.998716\t1\t0.998718\t0.706653\t0.666667\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.6.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "9\t981\t0\t0\t1\t1\t1\t1\t1\t1\t1\t1\t\n",
      "\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "2\t988\t0\t0\t1\t1\t1\t1\t1\t1\t1\t1\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.20.dist\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.35.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "15\t1937\t1\t0\t1\t0.999484\t0.9375\t1\t0.9375\t0.999488\t0.967996\t0.967742\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.34.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "12\t1636\t0\t5\t0.705882\t1\t1\t0.996953\t1\t0.996975\t0.838887\t0.827586\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.10.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "0\t943\t0\t3\t0\t1\t0\t0.996829\t1\t0.996829\t0\t0\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.14.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "36\t1140\t0\t0\t1\t1\t1\t1\t1\t1\t1\t1\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.4.dist\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "5\t1886\t0\t0\t1\t1\t1\t1\t1\t1\t1\t1\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.5.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "47\t3949\t2\t7\t0.87037\t0.999494\t0.959184\t0.998231\t0.959184\t0.997753\t0.91259\t0.912621\t\n",
      "\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "4\t1373\t1\t0\t1\t0.999272\t0.8\t1\t0.8\t0.999274\t0.894102\t0.888889\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.37.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "4\t940\t1\t1\t0.8\t0.998937\t0.8\t0.998937\t0.8\t0.997886\t0.798937\t0.8\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.28.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "13\t1115\t0\t0\t1\t1\t1\t1\t1\t1\t1\t1\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.44.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "10\t1881\t0\t0\t1\t1\t1\t1\t1\t1\t1\t1\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.13.dist\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.21.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "5\t983\t0\t2\t0.714286\t1\t1\t0.99797\t1\t0.99798\t0.844296\t0.833333\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.42.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "15\t1019\t0\t1\t0.9375\t1\t1\t0.99902\t1\t0.999034\t0.967771\t0.967742\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.39.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "7\t1217\t0\t1\t0.875\t1\t1\t0.999179\t1\t0.999184\t0.93503\t0.933333\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.18.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "4\t816\t0\t0\t1\t1\t1\t1\t1\t1\t1\t1\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.41.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "43\t1279\t3\t1\t0.977273\t0.99766\t0.934783\t0.999219\t0.934783\t0.996983\t0.954248\t0.955556\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.12.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "13\t1261\t0\t1\t0.928571\t1\t1\t0.999208\t1\t0.999216\t0.963242\t0.962963\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.19.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "64\t3577\t8\t6\t0.914286\t0.997768\t0.888889\t0.998325\t0.888889\t0.99617\t0.899549\t0.901408\t\n",
      "\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "0\t777\t0\t3\t0\t1\t0\t0.996154\t1\t0.996154\t0\t0\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.47.dist\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.3.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "182\t10223\t18\t17\t0.914573\t0.998242\t0.91\t0.99834\t0.91\t0.996648\t0.910575\t0.912281\t\n",
      "\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "183\t46137\t12\t28\t0.867299\t0.99974\t0.938462\t0.999393\t0.938462\t0.999137\t0.901752\t0.901478\t\n",
      "\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "1\t739\t0\t1\t0.5\t1\t1\t0.998649\t1\t0.99865\t0.706629\t0.666667\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.52.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "0\t779\t0\t1\t0\t1\t0\t0.998718\t1\t0.998718\t0\t0\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.27.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "16\t1998\t1\t1\t0.941176\t0.9995\t0.941176\t0.9995\t0.941176\t0.999008\t0.940676\t0.941176\t\n",
      "\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "0\t1127\t0\t1\t0\t1\t0\t0.999113\t1\t0.999113\t0\t0\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.17.dist\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.1.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "52\t2087\t5\t1\t0.981132\t0.99761\t0.912281\t0.999521\t0.912281\t0.997203\t0.944679\t0.945455\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.26.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "1506\t5478\t5\t32\t0.979194\t0.999088\t0.996691\t0.994192\t0.996691\t0.99473\t0.984562\t0.987865\t\n",
      "\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "152\t1557\t1\t1\t0.993464\t0.999358\t0.993464\t0.999358\t0.993464\t0.998831\t0.992822\t0.993464\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.0.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "1050\t28292\t106\t442\t0.703753\t0.996267\t0.908304\t0.984618\t0.908304\t0.981666\t0.79061\t0.793051\t\n",
      "\n",
      "\n",
      "Clustering C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.fasta.2.dist\n",
      "\n",
      "tp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "5138\t336935\t497\t1465\t0.778131\t0.998527\t0.911801\t0.995671\t0.911801\t0.994297\t0.839521\t0.83968\t\n",
      "\n",
      "It took 0 seconds to cluster\n",
      "Merging the clustered files...\n",
      "It took 0 seconds to merge.\n",
      "/******************************************/\n",
      "Running command: sens.spec(cutoff=0.03, list=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.list, column=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.dist, count=C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.uchime.pick.pick.count_table)\n",
      "\n",
      "NOTE: sens.spec assumes that only unique sequences were used to generate the distance matrix.\n",
      "\n",
      "label\tcutoff\tnumotus\ttp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "0.03\t0.03\t843\t8972\t2225938\t668\t2092\t0.810918\t0.9997\t0.930705\t0.999061\t0.930705\t0.998767\t0.868151\t0.866692\n",
      "\n",
      "It took 1 to run sens.spec.\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.sensspec\n",
      "\n",
      "/******************************************/\n",
      "Done.\n",
      "\n",
      "\n",
      "label\tcutoff\tnumotus\ttp\ttn\tfp\tfn\tsensitivity\tspecificity\tppv\tnpv\tfdr\taccuracy\tmcc\tf1score\n",
      "0.03\t0.03\t843\t8972\t2225938\t668\t2092\t0.8109\t0.9997\t0.9307\t0.9991\t0.9307\t0.9988\t0.8682\t0.8667\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.dist\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.list\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.sensspec\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "m.cluster.split(fasta='current', count='current', taxonomy='current', splitmethod='classify', taxlevel=4, cutoff=0.03)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Get consensus taxonomy of OTUs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > make.shared(list=current,count=current,label=0.03)\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.uchime.pick.pick.count_table as input file for the count parameter.\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.list as input file for the list parameter.\n",
      "0.03\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.shared\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "m.make.shared(list='current', count='current', label=0.03)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mothur > classify.otu(list=current,count=current,taxonomy=current)\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.denovo.uchime.pick.pick.count_table as input file for the count parameter.\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.list as input file for the list parameter.\n",
      "Using C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pds.wang.pick.taxonomy as input file for the taxonomy parameter.\n",
      "0.03\t843\n",
      "\n",
      "Output File Names: \n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.0.03.cons.taxonomy\n",
      "C:\\Users\\Haley\\Documents\\MiSeq\\MiSeq_SOP\\H1_Output\\stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.0.03.cons.tax.summary\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "m.classify.otu(list='current', count='current', taxonomy='current')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "file = input(\"You will need the shared and taxonomy output files for further analyses. Currently they are named stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.shared and stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.0.03.cons.taxonomy. Would you like to name them something more recognizable? Yes or No?\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "if file == 'Yes' or 'yes':\n",
    "    shared = input(\"What would you like to rename the shared file to?\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "m.rename.file(input = 'stability.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.shared', new = shared)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
